packages/block-editor/src/components/inserter/index.js

```diff
@@ -105,10 +129,47 @@ class Inserter extends Component {
 
 export default compose( [
 	withSelect( ( select, { rootClientId } ) => {
-		const { hasInserterItems } = select( 'core/block-editor' );
-
+		const { hasInserterItems, hasOneAllowedItem, getOneAllowedItem } = select( 'core/block-editor' );
+		const allowedBlock = getBlockType( getOneAllowedItem( rootClientId ) );
 		return {
 			hasItems: hasInserterItems( rootClientId ),
+			hasOneAllowedItem: hasOneAllowedItem( rootClientId ),
+			blockTitle: allowedBlock ? allowedBlock.title : '',
+		};
+	} ),
+	withDispatch( ( dispatch, ownProps, { select } ) => {
+		// eslint-disable-next-line no-restricted-syntax
+		function createIfOne() {
```

Comment:
`createIfOne` isn't very clear for someone with no prior context. How about `autoInsertBlock`? Or something else. :)

Reply:
I think I should rename them all including the selector